lib/internal/crypto/keys.js

```diff
@@ -525,14 +525,19 @@ function prepareAsymmetricKey(key, ctx) {
     return { format: kKeyFormatPEM, data: getArrayBufferOrView(key, 'key') };
   } else if (typeof key === 'object') {
     const { key: data, encoding, format } = key;
+
     // The 'key' property can be a KeyObject as well to allow specifying
     // additional options such as padding along with the key.
     if (isKeyObject(data))
       return { data: getKeyObjectHandle(data, ctx) };
     else if (isCryptoKey(data))
       return { data: getKeyObjectHandle(data[kKeyObject], ctx) };
-    else if (isJwk(data) && format === 'jwk')
+    else if (format === 'jwk') {
+      validateObject(data, 'key.key');
+      validateString(data.kty, 'key.key.kty');
```

Comment:
Is this check absolutely necessary? This changes the error code from `ERR_INVALID_ARG_VALUE` to `ERR_INVALID_ARG_TYPE`, which is technically a breaking change, I think.

Reply:
Great catch.